tests/lean/structInstExtraEta.lean

```diff
@@ -0,0 +1,11 @@
+/-! This test assures that that `toA` field is not eta expanded using the `with` pattern as it was
+  before #2478 -/
+
+structure A where
+
+structure B extends A
```

Comment:
Can you add a test with `structure C extends B` using `a with` too?

Whether the behavior has changed is less important than recording what the behavior actually is through the test.